### PR TITLE
WIP: Rework Science Instrument + fix some LineOfSight issues

### PR DESCRIPTION
**Pull request by andrewpaulreeves**

### CODE REVIEW

**Comment by andrewpaulreeves** [comment]
(The commit log is a horrendous mess as I only meant ot make a merge to my own fork/master. I'll try and tidy it up...)

File: soapy/numbalib/numbalib.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 import numpy
 import numba
 
-def bilinear_interp(data, xCoords, yCoords, interpArray, thread_pool=None, bounds_check=True):
+def bilinear_interp_new(data, xCoords, yCoords, interpArray, bounds_check=True):
```
Comment (matthewtownson) [inline soapy/numbalib/numbalib.py:14]: Does this function ever get used? I couldn't see it used from a quick scan

File: soapy/numbalib/numbalib.py [hunk 2]
```diff
@@ -21,7 +21,112 @@ def bilinear_interp(data, xCoords, yCoords, interpArray, thread_pool=None, bound
...
+    else:
+        bilinear_interp_numba_inbounds(data, xCoords, yCoords, interpArray)
+
+    return interpArray
+
+
+
+
+@numba.jit(nopython=True, parallel=True)
+def bilinear_interp_numba(data, xCoords, yCoords, interpArray):
```
Comment (matthewtownson) [inline soapy/numbalib/numbalib.py:39]: Again, couldn't see where this is used

**Comment by andrewpaulreeves** [comment]
will rebase this pull on my master branch

**Pull request closed without merging**